crates/light-client-prover/src/da_block_handler.rs

```diff
@@ -246,7 +246,7 @@ where
             l1_block.header().clone(),
             previous_lcp_output,
             self.network.get_l2_genesis_root(),
-            self.network.initial_batch_proof_method_ids(),
+            self.network.initial_batch_proof_method_ids().to_vec(),
```

Comment:
I wonder if we really have to pass a Vec into this method. A slice would be better to avoid OOMs.